sql/core/src/test/resources/sql-tests/inputs/predicate-functions.sql

```diff
@@ -0,0 +1,42 @@
+-- In
+select 1 in(1, 2, 3);
+select 1 in(2, 3, 4);
+select named_struct('a', 1, 'b', 2) in(named_struct('a', 1, 'b', 1), named_struct('a', 1, 'b', 3));
+select named_struct('a', 1, 'b', 2) in(named_struct('a', 1, 'b', 2), named_struct('a', 1, 'b', 3));
+
+-- EqualTo
+select 1 = 1;
+select 1 = '1';
+select 1.0 = '1';
+
+-- GreaterThan
+select 1 > '1';
```

Comment:
The following test cases are intended to test the end-to-end comparison between different types. It doesn't make much sense to re-write them with `checkEvaluation`. We'd have something like `checkEvaluation(GreaterThan(Cast(...), ...), true)` by manually added `Cast`.

Reply:
For the test cases that trigger implicit type casting, we can keep them here.